### ISSUE TITLE
Global variable detected as "unused" if used only from inside a function or from external included file

### ISSUE THREAD

**Issue by tmcdos** [slevomat/coding-standard#1866]
I have the following case - it is simplified from a real codebase and it does not look as trivial as this:
```php
$result = $pdo->query('SELECT permissions FROM client WHERE id = xxx');
$perm = $result->fetch(PDO::FETCH_NUM)[0];

function by_parent(int $id, bool $ordered = FALSE)
{
  global $perm;

  .....
  if ($perm['CAN_DO_X'])
  {
     ....
  }
  if ($perm['CAN_DO_Y'])
  {
     ....
  }
  ....
}
```
Variable **$perm** is properly initialized but only accessed from one or two functions in the file.
The sniff `Variables.UnusedVariable` reports variable **$perm** as `unused` but it should not.
Also, if the variable is not directly used in the current file but is accessed from another file which is included in the original - the sniff reports variable as unused.
And a third problem - it seems that the check for global/static keyword in function `isStaticOrGlobalVariable` assumes that each variable is declared on a separate line and failes if they are comma separated, e.g.
```php
global $var1, $var2, $var3;
```
only considers `$var1` as global - but not `$var2` and `$var3`.
